cda-main/src/main.rs

```diff
@@ -59,9 +67,101 @@ struct AppArgs {
     log_file_name: Option<String>,
 }
 
+#[derive(Error, Debug)]
+pub enum AppError {
+    #[error("Initialization failed `{0}`")]
+    InitializationFailed(String),
+    #[error("Resource error: `{0}`")]
+    ResourceError(String),
+    #[error("Connection error `{0}`")]
+    ConnectionError(String),
+    #[error("Configuration error `{0}`")]
+    ConfigurationError(String),
+    #[error("Data error `{0}`")]
+    DataError(String),
+    #[error("Error during execution `{0}`")]
+    RuntimeError(String),
+    #[error("Not found: `{0}`")]
+    NotFound(String),
+    #[error("Server error: `{0}`")]
+    ServerError(String),
+}
+
+impl From<TracingSetupError> for AppError {
+    fn from(value: TracingSetupError) -> Self {
+        match value {
+            TracingSetupError::ResourceCreationFailed(_) => ResourceError(value.to_string()),
+            TracingSetupError::SubscriberInitializationFailed(_) => {
+                InitializationFailed(value.to_string())
+            }
+        }
+    }
+}
+
+impl From<DoipGatewaySetupError> for AppError {
+    fn from(value: DoipGatewaySetupError) -> Self {
+        match value {
+            DoipGatewaySetupError::InvalidAddress(_) => ConnectionError(value.to_string()),
+
+            DoipGatewaySetupError::SocketCreationFailed(_)
+            | DoipGatewaySetupError::PortBindFailed(_) => InitializationFailed(value.to_string()),
+
+            DoipGatewaySetupError::InvalidConfiguration(_) => ConfigurationError(value.to_string()),
+
+            DoipGatewaySetupError::ResourceError(_) => ResourceError(value.to_string()),
+
+            DoipGatewaySetupError::ServerError(_) => ServerError(value.to_string()),
+        }
+    }
+}
+
+impl From<DiagServiceError> for AppError {
+    fn from(value: DiagServiceError) -> Self {
+        match value {
+            DiagServiceError::RequestNotSupported(_)
+            | DiagServiceError::BadPayload(_)
+            | DiagServiceError::ConnectionClosed
+            | DiagServiceError::UnexpectedResponse(_)
+            | DiagServiceError::EcuOffline(_)
+            | DiagServiceError::NoResponse(_)
+            | DiagServiceError::SendFailed(_)
+            | DiagServiceError::InvalidAddress(_)
+            | DiagServiceError::InvalidRequest(_)
+            | DiagServiceError::Timeout => ConnectionError(value.to_string()),
+
+            DiagServiceError::ParameterConversionError(_)
+            | DiagServiceError::UnknownOperation
+            | DiagServiceError::UdsLookupError(_)
+            | DiagServiceError::VariantDetectionError(_)
+            | DiagServiceError::AccessDenied(_)
+            | DiagServiceError::InvalidSession(_)
+            | DiagServiceError::Nack(_) => RuntimeError(value.to_string()),
+
+            DiagServiceError::InvalidSecurityPlugin => {
+                AppError::ConfigurationError(value.to_string())
```

Comment:
suggestion: Some cases prefix AppError::, some dont. It would be great if this is constistent. (Actually I think this is the only case _with_ the prefix)